### ISSUE TITLE
Hanging test

### ISSUE THREAD

**Issue by dkhalanskyjb** [Kotlin/kotlinx.coroutines#4259]
```
kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest.testOverSubscriptionStress[jvm]
```
was running for three hours on the CI: https://teamcity.jetbrains.com/buildConfiguration/KotlinTools_KotlinxCoroutines_Build/4856859

**Comment by dkhalanskyjb** [comment]
Thread dump:
```
Full thread dump OpenJDK 64-Bit Server VM (11.0.22+7-LTS mixed mode):

Threads class SMR info:
_java_thread_list=0x00007c36dc000c40, length=18, elements={
0x00007c372c028800, 0x00007c372c407800, 0x00007c372c40b800, 0x00007c372c420800,
0x00007c372c422800, 0x00007c372c424800, 0x00007c372c427000, 0x00007c372c429000,
0x00007c372c4c1000, 0x00007c372c9cd800, 0x00007c372c9dd000, 0x00007c372c9e9800,
0x00007c372d38a000, 0x00007c372cde2000, 0x00007c372cde3000, 0x00007c3678071800,
0x00007c3684026000, 0x00007c36dc001800
}

"Test worker" #1 prio=5 os_prio=0 cpu=20941.52ms elapsed=11647.71s tid=0x00007c372c028800 nid=0x18c3 waiting on condition  [0x00007c37349d1000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.22/Native Method)
	- parking to wait for  <0x00000000f3cbddb0> (a kotlinx.coroutines.BlockingCoroutine)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.22/LockSupport.java:234)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:98)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:69)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.testing.TestBase.runTest(TestBase.kt:139)
	at kotlinx.coroutines.testing.TestBase.runTest$default(TestBase.kt:131)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest.testOverSubscriptionStress(CoroutineSchedulerOversubscriptionTest.kt:58)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@11.0.22/Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@11.0.22/NativeMethodAccessorImpl.java:62)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.22/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@11.0.22/Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:112)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:40)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:60)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:52)
	at jdk.internal.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.22/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@11.0.22/Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

"Reference Handler" #2 daemon prio=10 os_prio=0 cpu=1.33ms elapsed=11647.68s tid=0x00007c372c407800 nid=0x18ca waiting on condition  [0x00007c37303fe000]
   java.lang.Thread.State: RUNNABLE
	at java.lang.ref.Reference.waitForReferencePendingList(java.base@11.0.22/Native Method)
	at java.lang.ref.Reference.processPendingReferences(java.base@11.0.22/Reference.java:241)
	at java.lang.ref.Reference$ReferenceHandler.run(java.base@11.0.22/Reference.java:213)

"Finalizer" #3 daemon prio=8 os_prio=0 cpu=0.50ms elapsed=11647.68s tid=0x00007c372c40b800 nid=0x18cb in Object.wait()  [0x00007c37302fe000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(java.base@11.0.22/Native Method)
	- waiting on <0x00000000c00013f0> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.22/ReferenceQueue.java:155)
	- waiting to re-lock in wait() <0x00000000c00013f0> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.22/ReferenceQueue.java:176)
	at java.lang.ref.Finalizer$FinalizerThread.run(java.base@11.0.22/Finalizer.java:170)

"Signal Dispatcher" #4 daemon prio=9 os_prio=0 cpu=0.31ms elapsed=11647.67s tid=0x00007c372c420800 nid=0x18cc runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Service Thread" #5 daemon prio=9 os_prio=0 cpu=0.06ms elapsed=11647.67s tid=0x00007c372c422800 nid=0x18cd runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #6 daemon prio=9 os_prio=0 cpu=14240.84ms elapsed=11647.67s tid=0x00007c372c424800 nid=0x18ce waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE
   No compile task

"C1 CompilerThread0" #8 daemon prio=9 os_prio=0 cpu=3022.59ms elapsed=11647.67s tid=0x00007c372c427000 nid=0x18cf waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE
   No compile task

"Sweeper thread" #9 daemon prio=9 os_prio=0 cpu=18.18ms elapsed=11647.67s tid=0x00007c372c429000 nid=0x18d0 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Common-Cleaner" #10 daemon prio=8 os_prio=0 cpu=7.36ms elapsed=11647.62s tid=0x00007c372c4c1000 nid=0x18d2 in Object.wait()  [0x00007c37140fe000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(java.base@11.0.22/Native Method)
	- waiting on <no object reference available>
	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.22/ReferenceQueue.java:155)
	- waiting to re-lock in wait() <0x00000000c0001948> (a java.lang.ref.ReferenceQueue$Lock)
	at jdk.internal.ref.CleanerImpl.run(java.base@11.0.22/CleanerImpl.java:148)
	at java.lang.Thread.run(java.base@11.0.22/Thread.java:829)
	at jdk.internal.misc.InnocuousThread.run(java.base@11.0.22/InnocuousThread.java:161)

"/127.0.0.1:33696 to /127.0.0.1:36517 workers" #12 prio=5 os_prio=0 cpu=10.45ms elapsed=11647.24s tid=0x00007c372c9cd800 nid=0x18d7 waiting on condition  [0x00007c370defd000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.22/Native Method)
	- parking to wait for  <0x00000000c0003860> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.22/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.22/AbstractQueuedSynchronizer.java:2081)
	at java.util.concurrent.ArrayBlockingQueue.put(java.base@11.0.22/ArrayBlockingQueue.java:367)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.submitToRun(TestWorker.java:207)
	- locked <0x00000000c0003698> (a org.gradle.api.internal.tasks.testing.worker.TestWorker)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:169)
	at jdk.internal.reflect.GeneratedMethodAccessor8.invoke(Unknown Source)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.22/DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(java.base@11.0.22/Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.22/ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.22/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.22/Thread.java:829)

"/127.0.0.1:33696 to /127.0.0.1:36517 workers Thread 2" #13 prio=5 os_prio=0 cpu=184.32ms elapsed=11647.23s tid=0x00007c372c9dd000 nid=0x18d8 waiting on condition  [0x00007c370ddfd000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.22/Native Method)
	- parking to wait for  <0x00000000c0041538> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.22/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.22/AbstractQueuedSynchronizer.java:2081)
	at org.gradle.internal.remote.internal.hub.queue.EndPointQueue.take(EndPointQueue.java:49)
	at org.gradle.internal.remote.internal.hub.MessageHub$ConnectionDispatch.run(MessageHub.java:322)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.22/ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.22/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.22/Thread.java:829)

"/127.0.0.1:33696 to /127.0.0.1:36517 workers Thread 3" #14 prio=5 os_prio=0 cpu=12.97ms elapsed=11647.23s tid=0x00007c372c9e9800 nid=0x18d9 runnable  [0x00007c370d9fe000]
   java.lang.Thread.State: RUNNABLE
	at sun.nio.ch.EPoll.wait(java.base@11.0.22/Native Method)
	at sun.nio.ch.EPollSelectorImpl.doSelect(java.base@11.0.22/EPollSelectorImpl.java:120)
	at sun.nio.ch.SelectorImpl.lockAndDoSelect(java.base@11.0.22/SelectorImpl.java:124)
	- locked <0x00000000c0015920> (a sun.nio.ch.Util$2)
	- locked <0x00000000c00156d0> (a sun.nio.ch.EPollSelectorImpl)
	at sun.nio.ch.SelectorImpl.select(java.base@11.0.22/SelectorImpl.java:141)
	at org.gradle.internal.remote.internal.inet.SocketConnection$SocketInputStream.read(SocketConnection.java:185)
	at com.esotericsoftware.kryo.io.Input.fill(Input.java:146)
	at com.esotericsoftware.kryo.io.Input.require(Input.java:178)
	at com.esotericsoftware.kryo.io.Input.readByte(Input.java:295)
	at org.gradle.internal.serialize.kryo.KryoBackedDecoder.readByte(KryoBackedDecoder.java:88)
	at org.gradle.internal.remote.internal.hub.InterHubMessageSerializer$MessageReader.read(InterHubMessageSerializer.java:64)
	at org.gradle.internal.remote.internal.hub.InterHubMessageSerializer$MessageReader.read(InterHubMessageSerializer.java:52)
	at org.gradle.internal.remote.internal.inet.SocketConnection.receive(SocketConnection.java:81)
	at org.gradle.internal.remote.internal.hub.MessageHub$ConnectionReceive.run(MessageHub.java:270)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.22/ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.22/ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(java.base@11.0.22/Thread.java:829)

"ForkJoinPool.commonPool-worker-7" #685 daemon prio=5 os_prio=0 cpu=8.31ms elapsed=11609.06s tid=0x00007c372d38a000 nid=0x1bd3 waiting on condition  [0x00007c370c1fe000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.22/Native Method)
	- parking to wait for  <0x00000000e025b4d8> (a java.util.concurrent.ForkJoinPool)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.22/LockSupport.java:194)
	at java.util.concurrent.ForkJoinPool.runWorker(java.base@11.0.22/ForkJoinPool.java:1628)
	at java.util.concurrent.ForkJoinWorkerThread.run(java.base@11.0.22/ForkJoinWorkerThread.java:183)

"DefaultDispatcher-worker-1 @coroutine#2378" #1136 daemon prio=5 os_prio=0 cpu=16.11ms elapsed=11601.09s tid=0x00007c372cde2000 nid=0x1d98 waiting on condition  [0x00007c36ed8fe000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.22/Native Method)
	- parking to wait for  <0x00000000f3cbdee0> (a java.util.concurrent.CountDownLatch$Sync)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.22/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(java.base@11.0.22/AbstractQueuedSynchronizer.java:885)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(java.base@11.0.22/AbstractQueuedSynchronizer.java:1039)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(java.base@11.0.22/AbstractQueuedSynchronizer.java:1345)
	at java.util.concurrent.CountDownLatch.await(java.base@11.0.22/CountDownLatch.java:232)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest.runAndCheck(CoroutineSchedulerOversubscriptionTest.kt:18)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest.access$runAndCheck(CoroutineSchedulerOversubscriptionTest.kt:9)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest$testOverSubscriptionStress$1$1$1$1.invokeSuspend(CoroutineSchedulerOversubscriptionTest.kt:65)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:829)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)

"DefaultDispatcher-worker-3 @coroutine#2379" #1138 daemon prio=5 os_prio=0 cpu=29.34ms elapsed=11601.09s tid=0x00007c372cde3000 nid=0x1d9a waiting on condition  [0x00007c36efbfe000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.22/Native Method)
	- parking to wait for  <0x00000000f3cbdee0> (a java.util.concurrent.CountDownLatch$Sync)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.22/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(java.base@11.0.22/AbstractQueuedSynchronizer.java:885)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(java.base@11.0.22/AbstractQueuedSynchronizer.java:1039)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(java.base@11.0.22/AbstractQueuedSynchronizer.java:1345)
	at java.util.concurrent.CountDownLatch.await(java.base@11.0.22/CountDownLatch.java:232)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest.runAndCheck(CoroutineSchedulerOversubscriptionTest.kt:18)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest.access$runAndCheck(CoroutineSchedulerOversubscriptionTest.kt:9)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest$testOverSubscriptionStress$1$1$1$1.invokeSuspend(CoroutineSchedulerOversubscriptionTest.kt:65)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:829)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)

"DefaultDispatcher-worker-2" #1140 daemon prio=5 os_prio=0 cpu=16.37ms elapsed=11601.09s tid=0x00007c3678071800 nid=0x1d9c waiting on condition  [0x00007c36efafe000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.22/Native Method)
	- parking to wait for  <0x00000000f3cbdee0> (a java.util.concurrent.CountDownLatch$Sync)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.22/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(java.base@11.0.22/AbstractQueuedSynchronizer.java:885)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(java.base@11.0.22/AbstractQueuedSynchronizer.java:1039)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(java.base@11.0.22/AbstractQueuedSynchronizer.java:1345)
	at java.util.concurrent.CountDownLatch.await(java.base@11.0.22/CountDownLatch.java:232)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest.runAndCheck(CoroutineSchedulerOversubscriptionTest.kt:18)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest.access$runAndCheck(CoroutineSchedulerOversubscriptionTest.kt:9)
	at kotlinx.coroutines.scheduling.CoroutineSchedulerOversubscriptionTest$testOverSubscriptionStress$1$1$1$1.invokeSuspend(CoroutineSchedulerOversubscriptionTest.kt:65)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:829)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)

"DefaultDispatcher-worker-4" #1141 daemon prio=5 os_prio=0 cpu=28.15ms elapsed=11601.09s tid=0x00007c3684026000 nid=0x1d9d waiting on condition  [0x00007c36f9bfe000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@11.0.22/Native Method)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.22/LockSupport.java:357)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.park(CoroutineScheduler.kt:855)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.tryPark(CoroutineScheduler.kt:803)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:751)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)

"Attach Listener" #1143 daemon prio=9 os_prio=0 cpu=0.21ms elapsed=0.10s tid=0x00007c36dc001800 nid=0x337b waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"VM Thread" os_prio=0 cpu=261.71ms elapsed=11647.68s tid=0x00007c372c3ff800 nid=0x18c9 runnable  

"GC Thread#0" os_prio=0 cpu=68.78ms elapsed=11647.71s tid=0x00007c372c041800 nid=0x18c4 runnable  

"GC Thread#1" os_prio=0 cpu=63.49ms elapsed=11646.86s tid=0x00007c36f0001000 nid=0x18da runnable  

"GC Thread#2" os_prio=0 cpu=64.06ms elapsed=11646.86s tid=0x00007c36f0002800 nid=0x18db runnable  

"GC Thread#3" os_prio=0 cpu=65.95ms elapsed=11646.86s tid=0x00007c36f0004000 nid=0x18dc runnable  

"G1 Main Marker" os_prio=0 cpu=1.02ms elapsed=11647.71s tid=0x00007c372c05d800 nid=0x18c5 runnable  

"G1 Conc#0" os_prio=0 cpu=69.22ms elapsed=11647.71s tid=0x00007c372c05f000 nid=0x18c6 runnable  

"G1 Refine#0" os_prio=0 cpu=3.44ms elapsed=11647.70s tid=0x00007c372c39d800 nid=0x18c7 runnable  

"G1 Young RemSet Sampling" os_prio=0 cpu=4265.85ms elapsed=11647.70s tid=0x00007c372c39f800 nid=0x18c8 runnable  
"VM Periodic Task Thread" os_prio=0 cpu=3399.61ms elapsed=11647.63s tid=0x00007c372c4b1800 nid=0x18d1 waiting on condition  

JNI global refs: 23, weak refs: 0
```

**Comment by dkhalanskyjb** [comment]
Duplicate of #3654
